tests/phpunit/tests/blocks/renderBlock.php

```diff
@@ -192,4 +192,70 @@ public function test_default_context_is_filterable() {
 
 		$this->assertSame( array( 'example' => 'ok' ), $provided_context[0] );
 	}
+
+	/**
+	 * Tests the behavior of the 'render_block_context' filter based on the location of the filtered block.
+	 *
+	 * @ticket 62046
+	 */
+	public function test_render_block_context_inner_blocks() {
+		$provided_context = array();
+
+		register_block_type(
+			'tests/context-provider',
+			array(
+				'provides_context' => array( 'example' ),
+			)
+		);
+
+		register_block_type(
+			'tests/context-consumer',
+			array(
+				'uses_context'    => array( 'example' ),
+				'render_callback' => static function ( $attributes, $content, $block ) use ( &$provided_context ) {
+					$provided_context = $block->context;
+
+					return '';
+				},
+			)
+		);
+
+		// Filter the context provided by the test block.
+		add_filter(
+			'render_block_context',
+			function ( $context, $parsed_block ) {
+				if ( isset( $parsed_block['blockName'] ) && 'tests/context-provider' === $parsed_block['blockName'] ) {
+					$context['example'] = 'ok';
+				}
+
+				return $context;
+			},
+			10,
+			2
+		);
+
+		// Test inner block context when the provider block is a top-level block.
+		do_blocks(
+			<<<HTML
+<!-- wp:tests/context-provider -->
```

Comment:
For the top level, it runs the filter inside `render_block`:

https://github.com/WordPress/wordpress-develop/blob/a78540b0881a195ec8488bb6bdf878114d75f8b8/src/wp-includes/blocks.php#L2110